nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/CaptureChangeMySQL.java

```diff
@@ -0,0 +1,928 @@
+            stateManager.setState(newStateMap, Scope.LOCAL);
+        }
+    }
+
+    protected void stop(StateManager stateManager) throws CDCException {
+        try {
+            binlogClient.disconnect();
+            if (eventListener != null) {
+                eventListener.stop();
+                binlogClient.unregisterEventListener(eventListener);
+            }
+            doStop.set(true);
+
+            // Update state with latest values
+            Map<String, String> newStateMap = new HashMap<>(stateManager.getState(Scope.LOCAL).toMap());
+
+            // Save current binlog filename and position to the state map
+            if (currentBinlogFile != null) {
+                newStateMap.put(BinlogEventInfo.BINLOG_FILENAME_KEY, currentBinlogFile);
+            }
+            newStateMap.put(BinlogEventInfo.BINLOG_POSITION_KEY, Long.toString(currentBinlogPosition));
+            newStateMap.put(EventWriter.SEQUENCE_ID_KEY, String.valueOf(currentSequenceId.get()));
+            stateManager.setState(newStateMap, Scope.LOCAL);
+            currentBinlogPosition = -1;
+
+        } catch (IOException e) {
+            throw new CDCException("Error closing CDC connection", e);
+        }
+        binlogClient = null;
+    }
+
+
+    /**
+     * Creates and returns a BinlogEventListener instance, associated with the specified binlog client and event queue.
+     *
+     * @param client A reference to a BinaryLogClient. The listener is associated with the given client, such that the listener is notified when
+     *               events are available to the given client.
+     * @param q      A queue used to communicate events between the listener and the NiFi processor thread.
+     * @return A BinlogEventListener instance, which will be notified of events associated with the specified client
+     */
+    BinlogEventListener createBinlogEventListener(BinaryLogClient client, LinkedBlockingQueue<RawBinlogEvent> q) {
+        return new BinlogEventListener(client, q);
+    }
+
+    BinaryLogClient createBinlogClient(String hostname, int port, String username, String password) {
+        return new BinaryLogClient(hostname, port, username, password);
+    }
+
+    /**
+     * Retrieves the column information for the specified database and table. The column information can be used to enrich CDC events coming from the RDBMS.
+     *
+     * @param key A TableInfoCacheKey reference, which contains the database and table names
+     * @return A TableInfo instance with the ColumnDefinitions provided (if retrieved successfully from the database)
+     */
+    protected TableInfo loadTableInfo(TableInfoCacheKey key) throws SQLException {
+        TableInfo tableInfo = null;
+        if (jdbcConnection != null) {
+            try (Statement s = jdbcConnection.createStatement()) {
+                s.execute("USE " + key.getDatabaseName());
+                ResultSet rs = s.executeQuery("SELECT * FROM " + key.getTableName());
```

Comment:
For the Distributed Cache comment, we could retrieve them from the RDBMS every time, but I was thinking the cache would be faster. If the difference is negligible, we can remove the use of the cache and just put in a processor-local one with a configurable cache size. What do you think?